### PR TITLE
Fix map buttons

### PR DESCRIPTION
**Pull request by evulins**
In this PR:

[- fixed showing proper view after clicking map buttons](https://trello.com/c/CKLLUPEv/1961-when-filter-is-clicked-hide-filter)
[- fixed active timelapse button after copy link](https://trello.com/c/R9tmQAX3/1964-timelapse-copy-link-paste-enter-link-into-new-browser-window-filter-button-should-be-white-timelapse-button-should-be-blue)
[- fixed setting active button](https://trello.com/c/KWjFthNd/1963-timelapse-click-x-timelapse-button-should-be-white-filter-button-should-be-blue-filter-should-be-displayed)
[- fixed timezone](https://trello.com/c/ZCvfB7dS/1965-timelapse-time-does-not-correspond-with-local-time-of-browser)

### CODE REVIEW

File: app/javascript/react/components/Modals/TimelapseModal/TimelapseComponent.tsx [hunk 1]
```diff
@@ -43,10 +43,10 @@ const TimelapseComponent: React.FC<
   const resetTimelapse = useCallback(() => {
     setCurrentStep(0);
     dispatch(setCurrentTimestamp(""));
-  }, []);
+  }, [dispatch]);
 
   const filteredTimestamps = useMemo(() => {
-    const now = moment.utc();
+    const now = moment();
```
Comment (Sonje) [inline app/javascript/react/components/Modals/TimelapseModal/TimelapseComponent.tsx:49]: Don't we use utc on the backend? Shouldn't we convert user's local time to utc then? Did the previous fitleredTImestamps solution work only for our timezone but not for Michael's?

**Review by Sonje — COMMENTED** [review]
I left one suggestion about huge if else block for the future, and I'm wondering about utc removal. I guess the conversion to UTC was causing the issues with filtering first and last timestamp, right?

I'll approve after retesting the issues reported in Slack thread

File: app/javascript/react/components/MapButtons/MapButtons.tsx [hunk 2]
```diff
@@ -28,10 +28,12 @@ const MapButtons = () => {
...
   } = useMapParams();
-  const [activeButtons, setActiveButtons] = useState<ButtonTypes[]>([
-    ButtonTypes.FILTER,
-  ]);
-  const [showFilters, setShowFilters] = useState(true);
+  const [activeButtons, setActiveButtons] = useState<ButtonTypes[]>([]);
+
+  const [activeCopyLinkButton, setActiveCopyLinkButton] = useState(false);
+
+  const showFilters = activeButtons.includes(ButtonTypes.FILTER);
```
Comment (JoannaThelema) [inline app/javascript/react/components/MapButtons/MapButtons.tsx:35]: `isFiltersButtonActive` for consistency with `isTimelapseButtonActive`

File: app/javascript/react/components/MapButtons/MapButtons.tsx [hunk 3]
```diff
@@ -40,51 +42,57 @@ const MapButtons = () => {
...
   return (
     <S.MapButtonsWrapper>
       <S.MapButtons>
         <MapButton
           title={t("navbar.filter")}
           image={filterIcon}
-          onClick={() => handleClick(ButtonTypes.FILTER)}
+          onClick={() => handleFilterClick()}
           alt={t("navbar.altFilter")}
           isActive={activeButtons.includes(ButtonTypes.FILTER)}
```
Comment (JoannaThelema) [inline app/javascript/react/components/MapButtons/MapButtons.tsx:97]: You can reuse `isFiltersButtonActive`

File: app/javascript/react/components/MapButtons/MapButtons.tsx [hunk 4]
```diff
@@ -106,21 +114,19 @@ const MapButtons = () => {
             <MapButton
               title={t("navbar.copyLink")}
               image={copyLinkIcon}
-              onClick={() => {}}
+              onClick={() => handleCopyLinkClick()}
               alt={t("navbar.altCopyLink")}
-              isActive={activeButtons.includes(ButtonTypes.COPY_LINK)}
+              isActive={activeCopyLinkButton}
```
Comment (JoannaThelema) [inline app/javascript/react/components/MapButtons/MapButtons.tsx:119]: I would unify the naming and have `activeFiltersButton` or rename this to `isCopy LinkButtonActive`

File: app/javascript/react/components/MapButtons/MapButtons.tsx [hunk 5]
```diff
@@ -106,21 +114,19 @@ const MapButtons = () => {
...
           showBelowButton
           onOpen={() => {
-            handleClick(ButtonTypes.COPY_LINK);
+            handleCopyLinkClick();
           }}
           onClose={() => {
-            setActiveButtons((prevState) =>
-              prevState.filter((type) => type !== ButtonTypes.COPY_LINK)
-            );
+            setActiveCopyLinkButton(false);
```
Comment (JoannaThelema) [inline app/javascript/react/components/MapButtons/MapButtons.tsx:129]: We can also unify this logic and have open/close in `handleCopyLinkClick()` similar to `handleTimelapseClick()`

**Review by Sonje — APPROVED** [review]

**Pull request merged**